server/tests/test_pipeline.py

```diff
@@ -82,3 +83,153 @@ def test_to_dict():
+                                {
+                                    "or_": [
+                                        {"column": "colM", "operator": "gt", "value": "__VOID__"},
+                                        {"column": "colN", "operator": "le", "value": "__VOID__"},
+                                        {
+                                            "and_": [
+                                                {
+                                                    "column": "colO",
+                                                    "operator": "eq",
+                                                    "value": "__VOID__",
+                                                },
+                                            ]
+                                        },
+                                        {
+                                            "column": "colP",
+                                            "operator": "in",
+                                            "value": [
+                                                "__VOID__",
+                                                "{{ __front_var_11__ }}",
+                                            ],
+                                        },
+                                        {
+                                            "column": "colQ",
+                                            "operator": "nin",
+                                            "value": [False, "{{ __front_var_10__ }}"],
+                                        },
+                                    ]
+                                },
+                            ]
+                        },
+                    ]
+                },
+            ]
+        },
+    }
+    composed_filter_step_and_all_voids = {
+        "name": "filter",
+        "condition": {
+            "and_": [
+                {"column": "colK", "operator": "gt", "value": "{{ __front_var_8__ }}"},
+                {"column": "colL", "operator": "le", "value": "{{ __front_var_9__ }}"},
+            ]
+        },
+    }
+
+    steps = [
+        {"domain": "foobar", "name": "domain"},
+        void_step,
+        composed_filter_step_or_and_,
+        simple_filter_step,
+        composed_filter_step_and_,
+        composed_filter_step_and_all_voids,
+    ]
+    variables = {
+        "__front_var_0__": "__VOID__",
+        "__front_var_1__": 32,
+        "__front_var_2__": "__VOID__",
+        "__front_var_3__": "TEST TEST",
+        "__front_var_4__": "__VOID__",
+        "__front_var_5__": "__VOID__",
+        "__front_var_6__": "VALUE",
+        "__front_var_7__": "TOTO",
+        "__front_var_8__": "__VOID__",
+        "__front_var_9__": "__VOID__",
+        "__front_var_10__": "__VOID__",
+        "__front_var_11__": True,
+    }
+
+    pipeline_with_variables = PipelineWithVariables(steps=steps)
+    pipeline = pipeline_with_variables.render(variables, renderer=_render_query)
```

Comment:
Using a private method from another lib does not seems to be a good idea. Having a simple jinja renderer defined here would make the test easier to understand

Reply:
True, good idea, added  this : https://github.com/ToucanToco/weaverbird/pull/1763/commits/9b0bb48ff4434966fd226401d633342e61219d1b